slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java

```diff
@@ -102,58 +107,101 @@ private void populateInterchangeData(final JsonGenerator jsonGenerator) {
-              }
-            });
+    if (watermark.getSlot() == null) {
+      LOG.warn(
+          "No block slot low watermark exists for {}, producing empty block listing",
+          validator.getPublicKey());
+    } else {
+
+      signedBlocksDao
+          .findAllBlockSignedBy(handle, validator.getId())
+          .forEach(
+              b -> {
+                if (b.getSlot().compareTo(watermark.getSlot()) >= 0) {
+                  final tech.pegasys.web3signer.slashingprotection.interchange.model.SignedBlock
+                      jsonBlock =
+                          new tech.pegasys.web3signer.slashingprotection.interchange.model
+                              .SignedBlock(b.getSlot(), b.getSigningRoot().orElse(null));
+                  try {
+                    mapper.writeValue(jsonGenerator, jsonBlock);
+                  } catch (final IOException e) {
+                    throw new UncheckedIOException(
+                        "Failed to construct a signed_blocks entry in json", e);
+                  }
+                } else {
+                  LOG.debug(
+                      "Ignoring block in slot {} for validator {}, as is below watermark",
+                      b.getSlot(),
+                      validator.getPublicKey());
+                }
+              });
+    }
 
     jsonGenerator.writeEndArray();
   }
 
   private void writeAttestations(
-      final Handle h, final Validator validator, final JsonGenerator jsonGenerator)
+      final Handle handle,
+      final SigningWatermark watermark,
+      final Validator validator,
+      final JsonGenerator jsonGenerator)
       throws IOException {
     jsonGenerator.writeArrayFieldStart("signed_attestations");
 
+    if (watermark.getSourceEpoch() == null || watermark.getTargetEpoch() == null) {
+      LOG.warn(
+          "Missing attestation low watermark for {}, producing empty attestation listing",
+          validator.getPublicKey());
+    }
+
     signedAttestationsDao
-        .findAllAttestationsSignedBy(h, validator.getId())
+        .findAllAttestationsSignedBy(handle, validator.getId())
         .forEach(
             a -> {
-              final tech.pegasys.web3signer.slashingprotection.interchange.model.SignedAttestation
-                  jsonAttestation =
-                      new tech.pegasys.web3signer.slashingprotection.interchange.model
-                          .SignedAttestation(
-                          a.getSourceEpoch(), a.getTargetEpoch(), a.getSigningRoot().orElse(null));
-              try {
-                mapper.writeValue(jsonGenerator, jsonAttestation);
-              } catch (final IOException e) {
-                throw new UncheckedIOException(
-                    "Failed to construct a signed_attestations entry in json", e);
+              if ((a.getSourceEpoch().compareTo(watermark.getSourceEpoch()) >= 0)
+                  && (a.getTargetEpoch().compareTo(watermark.getTargetEpoch()) >= 0)) {
```

Comment:
Should this just be target > 0? The attestation validator rule for hasTargetOlderThanWatermark excludes signing anything < watermark excluding the equals case. I believe this difference between source and target was from the interchange spec rules.